utils_linux.go

```diff
@@ -31,24 +31,26 @@ func loadFactory(context *cli.Context) (*libcontainer.Factory, error) {
 		return nil, err
 	}
 
-	intelRdtManager := libcontainer.IntelRdtFs
+	f, err := libcontainer.New(abs, libcontainer.IntelRdtFs)
+	if err != nil {
+		return nil, err
+	}
 
 	// We resolve the paths for {newuidmap,newgidmap} from the context of runc,
 	// to avoid doing a path lookup in the nsexec context. TODO: The binary
 	// names are not currently configurable.
-	newuidmap, err := exec.LookPath("newuidmap")
-	if err != nil {
-		newuidmap = ""
+	if newuidmap, err := exec.LookPath("newuidmap"); err == nil {
+		f.NewuidmapPath = newuidmap
```

Comment:
TBH; not sure I like this direction; with these changes, it looks like `loadFactory()` becomes the actual constructor, and not `New()`. I feel like either `New()` should do the constructing, or we should remove `New()` altogether if it's not really a constructor

Reply:
Yes, I guess I was going that way but fell short of the last step.

Reply:
OK, with a few more commits here and there I think I did it. PTAL @thaJeztah 

The only question is, should we call what's left of it a factory? I can certainly rename it but can't think of any good name -- suggestions welcome.

Reply:
> It would almost make sense to just libcontainer.Create(), libcontainer.Get(), libcontainer.List() (which internally can do a new()), adding root as argument.

This makes great sense (and we can get rid of `factory` entirely, so we don't have to think about naming it!).